### PR TITLE
Fix simplify Cos 0

### PR DESCRIPTION
**Pull request by zzc0430**
Fix simplify Cos 0

#### Fix cos(0, evaluate=False).simplify() will get 0 which is equal to 1

* functions
  * Fixed a bug with simplify Cos 0

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
I guess that this is a replacement for #24217. If that's the case then the other PR should be closed. Note that it isn't necessary to open a new PR just to make further changes though because you can just push more commits to the same PR.

**Comment by oscarbenjamin** [comment]
The authors test has failed because you need to add your git metadata to the .mailmap file.

File: sympy/functions/elementary/trigonometric.py [hunk 1]
```diff
@@ -1006,7 +1006,7 @@ def _eval_is_zero(self):
         if pi_mult:
             return fuzzy_and([(pi_mult - S.Half).is_integer, rest.is_zero])
         else:
-            return rest.is_zero
+            return (rest - S.Pi/2).is_zero
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/trigonometric.py:1009]: This is still not right:
```python
In [6]: e = cos(asin(-1, evaluate=False), evaluate=False)

In [7]: e
Out[7]: cos(asin(-1))

In [8]: e.evalf()
Out[8]: -0.e-189

In [9]: e.is_zero
Out[9]: False
```

**Comment by zzc0430** [comment]
> The authors test has failed because you need to add your git metadata to the .mailmap file.

I got it, thanks !

**Comment by oscarbenjamin** [comment]
This is fixed by #24219 now

**Pull request closed without merging**